include/event_manager/event_manager_priv.h

```diff
@@ -16,6 +16,11 @@
 extern "C" {
 #endif
 
+#if !CONFIG_X86
+#define _EM_FORCED_ALIGNMENT
+#else
+#define _EM_FORCED_ALIGNMENT __aligned(4)
+#endif
```

Comment:
I've seen this pattern before (tripping over the difference between a C array and a linker-aligned array).  It's a known gotcha, and IIRC already dealt with by the STRUCT_SECTION_ITERABLE feature, which I'm guessing this code should be using?

What's the purpose to constructing your own linker section, basically?  Seems like we have a utility for that already.

Reply:
Hi @andyross , the x86 thing is caused by cache line optimizations for larger data. Maybe zephyr should give a konfig option to control it and disable it by default.

As for using the sections, the idea was to have event definitions decentralized. When module is compiled in it registers intention for event handling by adding it's object to the section specific for this event. Everything is done on build time so it also make us keep event callback arrays on flash. We use orphan sections for that.

I did not introduce linker script as I thought it would mean centralization of events (event manager would need to know about all the events in the system, and create right entries in the linker script).
However there is one thing I realized after you comment. At the moment Zephyr gives you the possibility to add fragments of linker script (zephyr_linker_sources). What I think we would need to do is to define a small linker script file for each event where right section placement is explicitly defined. There are other benefits of this as we could use sorting.